flow/Error.h

```diff
@@ -75,38 +75,40 @@ Error systemErrorCodeToError();
 
 #undef ERROR
 #define ERROR(name, number, description) inline Error name() { return Error( number ); }; enum { error_code_##name = number };
+
 #include "error_definitions.h"
 
 //actor_cancelled has been renamed
 inline Error actor_cancelled() { return Error( error_code_operation_cancelled ); }
 enum { error_code_actor_cancelled = error_code_operation_cancelled };
 
 extern Error internal_error_impl( const char* file, int line );
-#define internal_error() internal_error_impl( __FILE__, __LINE__ )
+extern Error internal_error_impl(const char* msg, const char* file, int line);
+#define inernal_error_msg(msg) internal_error_impl(msg, __FILE__, __LINE__)
 
 extern bool isAssertDisabled( int line );
 //#define ASSERT( condition ) ((void)0)
 #define ASSERT(condition)                                                                                              \
 	do {                                                                                                               \
 		if (!((condition) || isAssertDisabled(__LINE__))) {                                                            \
-			throw internal_error();                                                                                    \
+			throw internal_error_impl(#condition, __FILE__, __LINE__);                                                 \
 		}                                                                                                              \
 	} while (false);
```

Comment:
Not related to your changes. But the ";" in this line should be removed.

Reply:
fixed.